### PR TITLE
Vending machines accept space cash.

### PR DESCRIPTION
**Pull request by Expletive**
:cl:
add: Thanks to the invention of "Bill slots", vending machines can now accept space cash.
/:cl:

I thought it was silly that vending machines couldn't accept space cash, so I've added this.

### CODE REVIEW

**Comment by Jalleo** [comment]
One Dollar from a service borg let you get the zippos then?

**Comment by Expletive** [comment]
I don't see why not?

**Comment by improvedname** [comment]
>roundstart

BORG GO SERVICE MODULE AND VEND A DOLLAR FOR ME SO I CAN BUY INSULATED GLOVES

>repeats between rounss

Borg salty in ooc

**Comment by meeeeme** [comment]
3. Obviously unreasonable or obnoxious orders (collect all X, do Y meaningless task) are a violation of Server Rule 1. The occurrence of such an attempt should be adminhelped and then disregarded.


1. Ordering a cyborg to pick a particular module. without an extreme need for a particular module.   or a prior agreement is both an unreasonable and an obnoxious order.

**Comment by InsaneHyena** [comment]
They operate by coin slots for a reason.

**Comment by Cyberboss** [comment]
Would help if cash values actually meant something

**Comment by octareenroon91** [comment]
Yes, we might be at the point to offer two tiers of premium item.

**Comment by FantasticFwoosh** [comment]
Just remove/replace the money printer from service borgs with fake money, or make it a additional emag real money printer. @improvedname

**Comment by improvedname** [comment]
Doesn't stop you from putting it in the ATM baw

**Comment by ma44** [comment]
They make 20 credits, it's not a problem

**Comment by lzimann** [comment]
@korphaeron

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -122,6 +123,8 @@
 	wires = null
 	qdel(coin)
 	coin = null
+	qdel(bill)
+	bill = null
```
Comment (Cyberboss) [inline code/game/machinery/vending.dm]: QDEL_NULL(bill)

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -443,16 +467,21 @@
...
 			return
-
-		coin.loc = loc
-		if(!usr.get_active_held_item())
-			usr.put_in_hands(coin)
-		to_chat(usr, "<span class='notice'>You remove [coin] from [src].</span>")
-		coin = null
-
+		if(coin)
+			coin.loc = loc
```
Comment (Cyberboss) [inline code/game/machinery/vending.dm]: remove this line

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -443,16 +467,21 @@
...
+			if(!usr.get_active_held_item())
+				usr.put_in_hands(coin)
+				to_chat(usr, "<span class='notice'>You remove [coin] from [src].</span>")
+				coin = null
```
Comment (Cyberboss) [inline code/game/machinery/vending.dm]: else
    coin.forceMove(get_turf(src))

unindent the to_chat and coin = null lines

File: code/game/machinery/vending.dm [hunk 4]
```diff
@@ -443,16 +467,21 @@
...
+		if(bill)
+			bill.loc = loc
```
Comment (Cyberboss) [inline code/game/machinery/vending.dm]: Remove this line

File: code/game/machinery/vending.dm [hunk 5]
```diff
@@ -443,16 +467,21 @@
...
+			if(!usr.get_active_held_item())
+				usr.put_in_hands(bill)
+				to_chat(usr, "<span class='notice'>You remove [bill] from [src].</span>")
+				bill = null
```
Comment (Cyberboss) [inline code/game/machinery/vending.dm]: else
    bill.forceMove(get_turf(src))

unindent the to_chat and bill= null lines

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by KorPhaeron** [comment]
Cool idea

**Comment by Really-Good-Soda-Flavor** [comment]
Maybe make it so a 100 credit bill is not the same as a 1 credit bill?

**Comment by Cyberboss** [comment]
If someone wants to implement proper cash values and prices, they can go right ahead

**Review by Cyberboss — COMMENTED** [review]
>Postmortem approval

**Pull request merged**